models/clm45/R/met2model.CLM45.R

```diff
@@ -19,207 +20,115 @@
-  Qair <- c(rep(Qair[1],toff),Qair)[1:slen]
-  U <- c(rep(U[1],toff),U)[1:slen]
-  V <- c(rep(V[1],toff),V)[1:slen]
-  Rain <- c(rep(Rain[1],toff),Rain)[1:slen]
-  pres <- c(rep(pres[1],toff),pres)[1:slen]
-  SW <- c(rep(SW[1],toff),SW)[1:slen]
-  LW <- c(rep(LW[1],toff),LW)[1:slen]
-  if(useCO2)  CO2 <- c(rep(CO2[1],toff),CO2)[1:slen]
- 
-### convert to CLM4.5 met variables
-
-#   netcdf \1999-01 {
-#     
-#     dimensions:
-#       
-#       scalar = 1 ;
-#     
-#     lon = 1 ;
-#     
-#     lat = 1 ;
-#     
-#     time = 744 ;
-#     
-#     variables:
-#       
-#       double time(time) ;
-#     
-#     time:long_name = "Time axis" ;
-#     
-#     time:units = "days since 1999-01-01 00:00:00" ;
-#     
-#     time:calendar = "noleap" ;
-#     
-#     double LONGXY(lat, lon) ;
-#     
-#     LONGXY:long_name = "longitude" ;
-#     
-#     LONGXY:units = "degrees E" ;
-#     
-#     double LATIXY(lat, lon) ;
-#     
-#     LATIXY:long_name = "latitude" ;
-#     
-#     LATIXY:units = "degrees N" ;
-#     
-#     double ZBOT(time, lat, lon) ;
-#     
-#     ZBOT:long_name = "observational height" ;
-#     
-#     ZBOT:units = "m" ;
-#     
-#     double EDGEW(scalar) ;
-#     
-#     EDGEW:long_name = "western edge in atmospheric data" ;
-#     
-#     EDGEW:units = "degrees E" ;
-#     
-#     double EDGEE(scalar) ;
-#     
-#     EDGEE:long_name = "eastern edge in atmospheric data" ;
-#     
-#     EDGEE:units = "degrees E" ;
-#     
-#     double EDGES(scalar) ;
-#     
-#     EDGES:long_name = "southern edge in atmospheric data" ;
-#     
-#     EDGES:units = "degrees N" ;
-#     
-#     double EDGEN(scalar) ;
-#     
-#     EDGEN:long_name = "northern edge in atmospheric data" ;
-#     
-#     EDGEN:units = "degrees N" ;
-#     
-#     double TBOT(time, lat, lon) ;
-#     
-#     TBOT:long_name = "temperature at the lowest atm level (TBOT)" ;
-#     
-#     TBOT:units = "K" ;
-#     
-#     double RH(time, lat, lon) ;
-#     
-#     RH:long_name = "relative humidity at the lowest atm level (RH)" ;
-#     
-#     RH:units = "%" ;
-#     
-#     double WIND(time, lat, lon) ;
-#     
-#     WIND:long_name = "wind at the lowest atm level (WIND)" ;
-#     
-#     WIND:units = "m/s" ;
-#     
-#     double FSDS(time, lat, lon) ;
-#     
-#     FSDS:long_name = "incident solar (FSDS)" ;
-#     
-#     FSDS:units = "W/m2" ;
-#     
-#     double FLDS(time, lat, lon) ;
-#     
-#     FLDS:long_name = "incident longwave (FLDS)" ;
-#     
-#     FLDS:units = "W/m2" ;
-#     
-#     double PSRF(time, lat, lon) ;
-#     
-#     PSRF:long_name = "pressure at the lowest atm level (PSRF)" ;
-#     
-#     PSRF:units = "Pa" ;
-#     
-#     double PRECTmms(time, lat, lon) ;
-#     
-#     PRECTmms:long_name = "precipitation (PRECTmms)" ;
-#     
-#     PRECTmms:units = "mm/s" ;
+  tstep = 86400/dt
   
+  doy <- rep(1:365,each=86400/dt)
+  if(as.numeric(year) %% 4 == 0){  ## is leap
+    doy <- rep(1:366,each=86400/dt)
```

Comment:
Again, use the `timestep.s` variable here.

More generally, you use `86400/dt` a lot, so I would pre-calculate it once somewhere above (with a descriptive label) and then use it everywhere else later.